agent/hcp/telemetry/otel_sink.go

```diff
@@ -20,8 +20,15 @@ import (
 // DefaultExportInterval is a default time interval between export of aggregated metrics.
 const DefaultExportInterval = 10 * time.Second
 
+// Enabler must be implemented as it is required to process metrics
+// Returning false will mean the sink is disabled, and will not accept metrics.
+type Enabler interface {
```

Comment:
@chapmanc Although I like this idea a lot, I think it might be neater to use a Disabler interface 😓 I know it's less intuitive in the telemetry_config_provider, but it's leading to a lot more complexity in the tests etc of the sink.

I think as functionality, to forcefully disable makes more sense to me starting off when using the sink, our use case is just a bit different here.